crates/nu-engine/src/eval.rs

```diff
@@ -591,6 +591,7 @@ pub fn eval_expression(
                 parts.push(eval_expression(engine_state, stack, expr)?);
             }
 
+            // TODO: Should this move to nu_engine::env::get_config?
```

Comment:
Yes

Reply:
Actually, this is a good place to watch out for performance: Running string interpolation in a hot loop would likely be slower with cloning the config. It might be worthwhile changing the `nu_engine::env::get_config` to output a `Cow`, or something like that—that could be a separate PR.

Reply:
`Cow<'a, T>` comes with a lifetime and optimizes for potential writes at the call. 

Sounds like a situation for the `Arc<T>` if it only needs reads (I assume we need `Send`/`Sync` for `par-each`?) or `Arc<Mutex<T>>` if we need a writeable config.

But yeah benchmarking is probably the first step with something like this.

Reply:
The problem is that `nu_engine::env::get_config()` can return either a reference, or an owned value, depending on whether it comes from the stack or the engine.